modules/host/default.nix

```diff
@@ -4,18 +4,41 @@
   self,
   microvm,
   netvm,
-}: {lib, ...}: {
-  imports = [
-    (import ./minimal.nix)
+}: {
+  lib,
+  config,
+  pkgs,
+  ...
+} @ args: let
+  cfg = config.ghaf.host.minification;
+in
+  with lib; {
+    imports = [microvm.nixosModules.host];
 
-    microvm.nixosModules.host
+    options.ghaf.host.minification = with lib; {
+      reduceProfile = mkEnableOption "reduced profile";
+      disableNetwork = mkEnableOption "disableing of network";
+      disableGetty = mkEnableOption "disabling of getty";
+      removeNix = mkEnableOption "nix tooling";
+      disableMicrovm = mkEnableOption "microvm";
```

Comment:
I'd propose we change this to `disableGuests` at this point. That way this will provide better semantics for the external users of the ghaf host module. It's a small change at this point but if we rename it later, it'll cause backward compatibility breaking change. We can also rename `mkEnableOption "microvm"` to `mkEnableOption "guestvms"` That will hide the internals of our microvm-based declarations and allow us to change the implementation if/when we have to use other implementations than `microvm.nix`.

Proposing also more vocal description - e.g. "Disables all ghaf reference guest virtual machines"

Reply:
I don't think that it's valuable to create elaborate descriptions for those options right now, because the will be changed in near refactoring.